File: Strings/LongestCommonPrefix/longestCommonPrefix.ts
function longestCommonPrefix(strs: string[]): string {
    strs.sort();
    const start = strs[0];
    const end = strs[strs.length - 1];
    let count = 0;

    while (count < start.length && count < end.length) {
        if (start[count] === end[count]) {
            count++;
        } else {
            break;
        }
    }
    return start.substring(0, count);
}

File: Strings/LongestCommonPrefix/longestCommonPrefix.py
class Solution:
    def longestCommonPrefix(self, strs: list[str]) -> str:
        strs.sort()  # important step
        start, end = strs[0], strs[-1]
        count = 0
        
        while count < len(start) and count < len(end):
            if start[count] == end[count]:
                count += 1
            else:
                break
        return start[:count]
